core/camel-api/src/main/java/org/apache/camel/CamelContext.java

```diff
@@ -412,6 +412,19 @@
      */
     Endpoint getEndpoint(String uri);
 
+    /**
+     * Resolves the given uri to an {@link Endpoint} of the specified type.
+     *
+     * @param fmt the string format
+     * @param args the arguments referenced by the format
+     * @return the endpoint
+     *
+     * @see #getEndpoint(String)
+     */
+    default Endpoint getEndpointF(String fmt, Object... args) {
```

Comment:
I know default interface methods are nice, but can we move this to AbstractCamelContext so we dont end up cluttering this interface with code.

Reply:
I think I will remove it as I was supposed to find the pattern to create URIs with string format more often but in fact I think it is not so popular